shadow_sampler.py

```diff
@@ -0,0 +1,104 @@
+import os.path as osp
+
+import argparse
+import torch
+import torch.nn.functional as F
+from torch_geometric.datasets import Flickr
+from torch_geometric.loader import GraphSAINTRandomWalkSampler
+from torch_geometric.loader import ShaDowKHopSampler
+from torch_geometric.nn import GraphConv
+from torch_geometric.utils import degree
+
+path = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data', 'Flickr')
+dataset = Flickr(path)
+data = dataset[0]
+row, col = data.edge_index
+data.edge_weight = 1. / degree(col, data.num_nodes)[col]  # Norm by in-degree.
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--use_normalization', action='store_true')
+args = parser.parse_args()
+
+
+indices = data.train_mask.nonzero()
+index = torch.flatten(indices)
+loader = ShaDowKHopSampler(data, depth=2, num_neighbors=5, node_idx=index)
+
+
+class Net(torch.nn.Module):
```

Comment:
Let's make direct use of `torch_geometric.nn.GraphSAGE` to reduce code size:, see https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.models.GraphSAGE.

Reply:
Okay. I have changed it to SAGEConv.